### PR TITLE
Fix transition rates with lasers on multiple transitions

### LINKED ISSUE #15: Rates.get_stim() broken for multiple transitions
Rates.get_stim() using lasers on multiple different transitions is broken.

It seems like the culprit  is that `Gamma` and `GammaJ` are overwritten in the transitions loop
https://github.com/OxfordIonTrapGroup/ion_phys/blob/da3c5ef9fe868aa297426506801e9f02a77c2f67/ion_phys/rate_equations.py#L18-L24
https://github.com/OxfordIonTrapGroup/ion_phys/blob/da3c5ef9fe868aa297426506801e9f02a77c2f67/ion_phys/rate_equations.py#L61-L66

### PR DESCRIPTION
**Pull request by pathfinder49**
Fixes #15

### CODE REVIEW

File: ion_phys/rate_equations.py [hunk 1]
```diff
@@ -59,11 +59,12 @@ def get_stim(self, lasers):
             delta_lu = Eu - El
 
             # Total scattering rate out of each state
-            GammaJ = GammaJ[upper_states]
-            GammaJ = np.repeat(GammaJ, n_lower).reshape(n_upper, n_lower).T
-            GammaJ2 = np.power(GammaJ, 2)
+            GammaJ_reduced = GammaJ[upper_states]
```
Comment (hartytp) [inline ion_phys/rate_equations.py:62]: nit pick: `_subspace` or similar would have been more in keeping with the notation used in the rest of the code.

Comment (hartytp) [reply ion_phys/rate_equations.py:62]: or just `_subs` to save characters.

File: ion_phys/rate_equations.py [hunk 2]
```diff
@@ -72,7 +73,8 @@ def get_stim(self, lasers):
                 for laser in [laser for laser in _lasers if laser.q == q]:
                     delta = delta_lu - laser.delta
                     I = laser.I
-                    R += GammaJ2/(4*np.power(delta, 2) + GammaJ2)*I*(Q*Gamma)
+                    R += GammaJ2/(4*np.power(delta, 2) + GammaJ2)
+                    R *= I*(Q*Gamma_reduced)
```
Comment (hartytp) [inline ion_phys/rate_equations.py:77]: why move this onto a new line?

File: ion_phys/tests/test_stim.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from ion_phys.ions.ca43 import Ca43, ground_level, P12, D32
+from ion_phys import Laser
+from ion_phys.rate_equations import Rates
+import unittest
+
+class TestGamma(unittest.TestCase):
+    def test_multi_transition(self):
+        """Test for no errors with lasers on multiple transitions"""
```
Comment (hartytp) [inline ion_phys/tests/test_stim.py:8]: "Test for no errors" is an odd description. Maybe include a URL for the relevant issue?

File: ion_phys/rate_equations.py (hunk 2 above)
Comment (pathfinder49) [reply ion_phys/rate_equations.py:77]: It's a long line. Broke it up for flake8. Messed up the maths though, will fix.

**Comment by pathfinder49** [comment]
Can this be merged?

**Comment by hartytp** [comment]
Yes. Not sure if this is really the right way of using GH, but when I leave feedback but still approve I mean "fix these uncontroversial/small points and then merge"

**Pull request merged**